### ISSUE TITLE
Null setting group causes fatal error in src/Data/DataSource.php

### ISSUE THREAD

**Issue by maurocolella** [wp-graphql/wp-graphql#2553]
### Description

Still on master (WP GraphQL 1.11.2 as of today): this line
https://github.com/wp-graphql/wp-graphql/blob/develop/src/Data/DataSource.php#L438

```
$group = self::format_group_name( $setting['group'] );
```

Can error out, potentially silently, if for some reason the setting group is null. This should be reproducible using WP Media Cloud, which seems to have such a null-labeled settings group in version 4.2.37:
https://wordpress.org/plugins/ilab-media-tools/

For now, our workaround has simply been to test for emptiness.
```
if( ! empty( $setting['group'] ) ) $group = self::format_group_name( $setting['group'] );
```

### Steps to reproduce

See above.

### Additional context

In our case, we use external components to display a stack trace on any error, which means that the issue effectively breaks display on some screens. For ex. when configuring fields with ACF.

### WPGraphQL Version

1.11.2

### WordPress Version

5.7.2

### PHP Version

7.4

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have disabled ALL plugins except for WPGraphQL.

- [ ] Yes
- [X] My issue is with compatibility with a specific WordPress plugin, and I have listed all my installed plugins (and version info) above.

**Comment by justlevine** [comment]
Hey @maurocolella 

I've added a fix in #2944 to prevent a fatal error for an unset group.

That said, I've been unable to actually replicate a scenario where `$settings['group']` is unset. Using `register_setting()` with an empty group defaults to the (deprecated) `misc` group, and as of today at least (WP 6.3.1 + WPGraphQL 1.16.0 + Media Cloud 4.5.24) seems to be working fine as well. Perhaps it was addressed in one of the version updates in the time this ticket has been open 🤷.
